worlds/dark_souls_3/Locations.py

```diff
@@ -688,6 +688,116 @@ def get_name_to_id() -> dict:
         [DS3LocationData(f"Undead Bone Shard #{i + 1}", "Undead Bone Shard", DS3LocationCategory.HEALTH) for i in range(10)],
 }
 
+# In addition to these hand-authored location groups, every region automatically
+# has a location group added in the loop below.
+location_name_groups: Dict[str, Set[str]] = {
+    # Locations that are very obvious: boss/miniboss drops (one per boss), keys,
+    # items unlocked by keys, visibly guarded items, or items at the end of a
+    # bossless area. Intended for players without broad DS3 knowledge or who
+    # want fewer, more obvious checks.
+    "Prominent Locations": frozenset([
+        "FS: Tower Key",
+        "FS: Uchigatana",
+        "FSBT: Covetous Silver Serpent Ring",
+        "FSBT: Fire Keeper Robe",
+        "FSBT: Fire Keeper Gloves",
+        "FSBT: Fire Keeper Skirt",
+        "FSBT: Estus Ring",
+        "FSBT: Fire Keeper Soul",
+        "HWL: Basin of Vows",
+        "HWL: Small Lothric Banner",
+        "HWL: Cell Key",
+        "HWL: Soul of Boreal Valley Vordt",
+        "HWL: Soul of the Dancer",
+        "US: Soul of the Rotted Greatwood",
+        "US: Flynn's Ring", # Reward for killing the Fire Demon
```

Comment:
Flynn's Ring is up on top of the building with the evangelists. Just before you drop off onto the tower with the Cloranthy Ring.

The reward for defeating the fire demon is only a Fire Gem.

Reply:
Yeah, maybe my comment here is unclear—I mean that this is _included in this list_ so that the player can get a progression item as a reward for defeating the Fire Demon and exploring that area.

Reply:
I see, these are prebuilt groups to put into priority and excluded locations.

In that case I have a different comment: this is a _lot_ of locations for priority locations. DS3 adds at most 17 progression items to the pool, and we're looking at 63 locations here (for priority locations). That's fine for large multiworlds that include games other than DS3, but it has the potential to eat up most of the pool's progression items with single digit player counts, especially if the other games also don't add many progression items.

Of course, it's on the player to opt in, so for it to be a problem, the player has to be negligent with their own options, but I think it's best not to enable that behavior if we can help it.

Splitting the group up, potentially into "tiers", seems like a good option to me. My intuition says `AL: Cinders of a Lord - Aldritch`, `CC: Soul of High Lord Wolnir`, `CD: Spider Shield`, and `ID: Jailer's Key King` are in progressively less significant categories. But I'm curious what Marech others think too.

Reply:
Oh, I didn't realize that priority items would pull in _all_ the progress items from every game. I was expecting it to be more like "don't place any progress items in places that aren't these locations."

I think splitting it up into tiers makes sense.

Reply:
I've updated this to use logical groups and narrowed "Progression Locations" to _only_ cover boss souls and progression items.